### PR TITLE
Add BLS signature verification for mainnet -> schain messages

### PR DESCRIPTION
**Pull request by DimaStebaev**

### CODE REVIEW

File: proxy/contracts/schain/bls/SkaleVerifier.sol [hunk 1]
```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+/*
+    SkaleVerifier.sol - SKALE Manager
+    Copyright (C) 2018-Present SKALE Labs
```
Comment (payvint) [inline proxy/contracts/schain/bls/SkaleVerifier.sol]: Need to change year and author

File: proxy/contracts/test/MessageProxyForSchainWithoutSignature.sol [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 
 /**
- *   IContractManager.sol - Interface of ContractManager Contract
+ *   MessageProxyForSchain.sol - SKALE Interchain Messaging Agent
  *   Copyright (C) 2019-Present SKALE Labs
```
Comment (payvint) [inline proxy/contracts/test/MessageProxyForSchainWithoutSignature.sol]: Need to change year and author

File: test/accounts.json [hunk 3]
```diff
@@ -1 +1 @@
...
...0186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"}}},"private_keys":{"0x0bf1b46b1ee16e569294656d2960df30f7f4a8d5":"b90e1989d1c47d084952237c2a2bb21f484d2ed5967735c3ca5f798ca4bfb2db","0xa8b9d846cac548780694ca525793fcab14972f25":"de12c9fb08fd2bfc15defef6062e96fb081f140ac461405694ba0efa85ed89df","0x0ab368f0a816246bda43b207f993c69743a489d5":"f2345978e8b2a03b58910582b3d3c36396260842b60cd3f100fb7fbee3007e68","0x07869f4350411a2ff5092feb0125c76c81221ba1":"9efd28592e5bddef9a6f8df56eec9d55929ab3dc051935c658aeff8d3cfda223","0x2dbb05d7103ca302734278c8a32b507336a25a4e":"314f55d42ca93ace198a4855ce9647ffb6c5881c5333c7441bafc7d15902d38c","0xa780e330d91837827c04020e83206ce00678d14e":"eeaf4500497026591b622ad29adffb65f9b10066b45f215383bdb140e00377a1","0x86cbcdb90051fd59d0f4fa753ec8bd282edc8427":"f04db0bb4031020c2011a44bde9cc775c8303323e5b3bd7ca5e292d331f3dcbe","0x1dbc928ab8b206c31f336a9484a331f420a74884":"84fa97896884b615ff8a64afa03f919767ece931027f144d6fdaa9ece29923b9","0xd9015224b560462f68c452819494892b3f925c13":"0237f53245107231d4527217d59e59a331967c1f1bbbcd2daf8a1d1ad4ad5fd4","0xcccdf08a2f6583997f8dc89532e3f162db0ba3c9":"ad14943cb34d5dfacf9945d0631681b7269f333e37076642c69c43516e936b3a"}}
```
Comment (payvint) [inline test/accounts.json]: Looks like we need to delete it

File: proxy/.solcover.js [hunk 4]
```diff
@@ -8,8 +8,8 @@ module.exports = {
     ],
     providerOptions: {
         default_balance_ether: 3000, //000000000000000000,
-        callGasLimit: 10000000,
-        gasLimit: 0xfffffffffff,
+        callGasLimit: "10000000",
```
Comment (payvint) [inline proxy/.solcover.js:11]: Maybe let's add "" everywhere in config files? (.solcover.js, truffle-config.js)

**Review by payvint — CHANGES_REQUESTED** [review]
Looks pretty good - let's have some little doc and syntax changes

File: proxy/.solcover.js (hunk 4 above)
Comment (DimaStebaev) [reply proxy/.solcover.js:11]: I did but not everything supports value in string.

**Review by payvint — APPROVED** [review]
Nice!

**Review by sergiy-skalelabs — APPROVED** [review]

**Pull request merged**